SplitFiles/base.h

```diff
@@ -0,0 +1,70 @@
+#ifndef _DEFINE_BASE_H
+#define _DEFINE_BASE_H
+
+// 1.) Forward declaration: Here is the name of the structure
+// but member-details are omitted.
+struct BreadBinStruct;
+
+// 2.) typedef of the structure
+// Still no details on the members.
+typedef struct BreadBinStruct  BreadBin;
+
+struct BreadTypeStruct {
+    unsigned char     letter;
+    int               thermalMass;      //This mass
+    int               massVariability;  //Plus or minus this amount
+    int               moisture;         //This moisture
+    int               moistureVariability;  //Plus or minus...
+    int               cost;
+    int               callProb;         //Actually an integer, higher number = more likely
+    char*             desc;
+};
+typedef struct BreadTypeStruct BreadType;
+
+struct BreadBinStruct {
+    int probTotal;
+    BreadType** breadTypes;                    //will be an array of 26 pointers 
+                                               //       where (BreadTypeStruct.letter - 'A') is the index
+};
+
+struct GameParameters_Struct;
+
+typedef struct GameParameters_Struct GameParameters;
+
+// Forward declaration
+struct GameComponent_Struct;
+
+// typedef of the structure
+typedef struct GameComponent_struct  GameComponent;
+
+// 3.) The parameter to the function is listed, using the definition
+// from step 2.)  (note: Still no details on the members yet)
+//typedef void (*GameComponentFunc)(GameComponent* input, GameParameters* params);
+typedef void (*GameComponentFunc)(GameComponent*, GameParameters*);
+
+struct GameParameters_Struct {
+  int       ticks; // How many times has main loop run
+  long       score; // What's the score
```

Comment:
previous comment may have been lost - should put the long as the first entry here as a memory optimisation (avoids fragmentation - possibly)

Reply:
Made both these tweaks, thankfully code now less reliant on parameter order for struct population!